app/controllers/user.rb

```diff
@@ -0,0 +1,115 @@
+require 'erb'
+require 'sinatra/flash'
+require 'pony'
+require 'json'
+
+module  Sinatra
+  module App
+    module SignUp
+
+      def self.registered(app)
+
+        app.get '/signup' do
+          @title = 'SignUp Form'
+          erb :signup
+        end
+
+
+        app.post "/unauthenticated" do
+          session[:return_to] = env['warden.options'][:attempted_path] if session[:return_to].nil?
+          puts env['warden.options'][:attempted_path]
+          flash[:error] = env['warden.options'][:message] || 'Invalid email or password.'
+          redirect 'login'
+        end
+
+        app.get '/login' do
+          @title = 'Login Form'
+          erb :login
+        end
+        app.post '/login' do
+          if  env['warden'].authenticate(:admin)
+            redirect '/admin', flash[:success] = 'Successfully logged in'
+          elsif  env['warden'].authenticate!
+            redirect '/dashboard', flash[:success] = 'Successfully logged in'
+          else
+            redirect '/login', flash[:error] = 'Incorrect email or password'
+          end
+        end
+
+        app.get '/logout' do
+          warden_handler.raw_session.inspect
+          warden_handler.logout
+          flash[:success] = 'Successfully logged out'
+          redirect '/'
+
+        end
+
+        app.get '/admin' do
+          check_admin_authentication
+          @topics = Topic.order('updated_at  DESC, id  DESC ')
+          @resources = Resource.order('updated_at  DESC, id  DESC ')
+          @users = User.order('updated_at  DESC, id  DESC ')
```

Comment:
In your admin view that is rendered by this route handler, you are only showing the **count** of these items(topics, resources and users). As such, **retrieving** all items and **ordering** them is not an optimal approach for what you are trying to achieve.

If you only need the count of a resource, you could simply use:
```ruby
@user_count = User.count
```